internal/providers/pluginfw/resources/qualitymonitor/resource_quality_monitor.go

```diff
@@ -0,0 +1,214 @@
+package qualitymonitor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/catalog_tf"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+const qualityMonitorDefaultProvisionTimeout = 15 * time.Minute
+
+var _ resource.Resource = &QualityMonitorResource{}
+
+func ResourceQualityMonitor() resource.Resource {
+	return &QualityMonitorResource{}
+}
+
+func waitForMonitor(ctx context.Context, w *databricks.WorkspaceClient, monitor *catalog.MonitorInfo) diag.Diagnostics {
+	monitorName := monitor.TableName
+	err := retry.RetryContext(ctx, qualityMonitorDefaultProvisionTimeout, func() *retry.RetryError {
+		monitor, err := w.QualityMonitors.GetByTableName(ctx, monitorName)
+		if err != nil {
+			return retry.NonRetryableError(err)
+		}
+
+		switch monitor.Status {
+		case catalog.MonitorInfoStatusMonitorStatusActive:
+			return nil
+		case catalog.MonitorInfoStatusMonitorStatusError, catalog.MonitorInfoStatusMonitorStatusFailed:
+			return retry.NonRetryableError(fmt.Errorf("monitor status retrund %s for monitor: %s", monitor.Status, monitorName))
+		}
+		return retry.RetryableError(fmt.Errorf("monitor %s is still pending", monitorName))
+	})
+	if err != nil {
+		return diag.Diagnostics{diag.NewErrorDiagnostic("Failed to get Monitor", err.Error())}
```

Comment:
```suggestion
		return diag.Diagnostics{diag.NewErrorDiagnostic("failed to get monitor", err.Error())}
```
nit: errors should start with lower-case letters